test/rules/src/https_everywhere_checker/rules.py

```diff
@@ -179,6 +181,28 @@ def _determineTestApplication(self):
 				self.determine_test_application_run = True
 		return self.test_application_problems
 
+	def getTargetValidityProblems(self):
+		"""Verify that each target has a valid TLD in order to prevent problematic rewrite
+			 as stated in EFForg/https-everywhere/issues/10877. In particular, 
+			 right-wildcard target are ignored from this test.
+
+			 Returns an array of strings reporting any coverage problems if they exist,
+			 or empty list if coverage is sufficient.
+			 """
+		problems = self._determineTestApplication()
+		# Next, make sure each target has a valid TLD
+		for target in self.targets:
+			# Ignore right-wildcard targets
+			if target.endswith(".*"):
+				continue
```

Comment:
They are valid but I am not convinced it is a good idea to include them in HTTPS-Everywhere rulesets.

[I had a look at rulesets including IP addresses a month ago](https://github.com/EFForg/https-everywhere/pulls?q=is%3Apr+ip+is%3Aclosed+author%3ABisaloo) and most of them were oudated. Best case scenario was the IP address is not responding on port 80. But worst case scenario, HTTPS-Everywhere was hijacking the DNS resolution and redirecting to the wrong domain, which is quite bad (eg [here](https://github.com/EFForg/https-everywhere/pull/10337)).

Adding IP addresses heavily increases the maintenance load without any clear benefit (very very few people will access a website by typing the IP address instead of the domain name).